diff --git a/better_exposed_filters.module b/better_exposed_filters.module
index a11beb5..379c84f 100644
--- a/better_exposed_filters.module
+++ b/better_exposed_filters.module
@@ -194,7 +194,7 @@ function better_exposed_filters_preprocess_views_view(&$variables) {
   $filters = array();
   foreach ($variables['view']->filter as $filter) {
     if ($filter->options['exposed']) {
-      $identifier = $filter->options['is_grouped'] ? $filter->options['group_info']['identifier'] : $filter->options['expose']['identifier'];
+      $identifier = !empty($filter->options['is_grouped']) ? $filter->options['group_info']['identifier'] : $filter->options['expose']['identifier'];
       $filters[$identifier] = array(
         'required' => $filter->options['expose']['required'] ? TRUE : FALSE,
       );
diff --git a/better_exposed_filters_exposed_form_plugin.inc b/better_exposed_filters_exposed_form_plugin.inc
index 7298658..98dac3d 100644
--- a/better_exposed_filters_exposed_form_plugin.inc
+++ b/better_exposed_filters_exposed_form_plugin.inc
@@ -1078,7 +1078,7 @@ dateFormat: "dd-mm-yy"
 
       // Form element is designated by the element ID which is user-
       // configurable.
-      $filter_key = 'filter-' . ($filters[$label]->options['is_grouped'] ? $filters[$label]->options['group_info']['identifier'] : $label);
+      $filter_key = 'filter-' . (!empty($filters[$label]->options['is_grouped']) ? $filters[$label]->options['group_info']['identifier'] : $label);
       $filter_id = $form['#info'][$filter_key]['value'];
 
       // Token replacement on BEF Description fields.
